ui/apps/platform/src/Components/PatternFly/MenuDropdown.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { ReactElement, ReactNode, useState } from 'react';
+import {
+    Dropdown,
+    DropdownList,
+    DropdownPopperProps,
+    MenuToggle,
+    MenuToggleElement,
+    MenuToggleProps,
+} from '@patternfly/react-core';
+
+type MenuDropdownProps = {
+    children: ReactNode;
+    toggleText: string;
+    toggleId?: string;
+    toggleClassName?: string;
+    toggleVariant?: MenuToggleProps['variant'];
+    onSelect?: (event?: React.MouseEvent<Element, MouseEvent>, value?: string | number) => void;
+    isDisabled?: boolean;
+    popperProps?: DropdownPopperProps;
+};
+
+// TODO: Reuse this for the Violations Page Bulk Actions
+function MenuDropdown({
+    children,
+    toggleText,
+    toggleId = 'menu-dropdown',
+    toggleClassName,
+    toggleVariant = 'default',
+    onSelect,
+    isDisabled = false,
+    popperProps,
+}: MenuDropdownProps): ReactElement {
+    const [isOpen, setIsOpen] = useState(false);
+
+    function onToggleClick() {
+        setIsOpen(!isOpen);
+    }
+
+    function onSelectHandler(event) {
+        setIsOpen(false);
+        onSelect?.call(event);
```

Comment:
My bad. This actually won't work correctly. Forgot about those apply, call, bind functions 😢